scipy/stats/_sensitivity_analysis.py

```diff
@@ -542,28 +590,34 @@ def sobol_indices(
         A, B = sample_A_B(n=n, dists=dists, random_state=random_state)
         AB = sample_AB(A=A, B=B)
 
-        f_A = np.asarray(func(A))
+        f_A = np.atleast_2d(func(A))
 
-        if f_A.shape[0] != n or f_A.shape[1] < 1:
+        if f_A.shape[1] != n or f_A.shape[0] < 1:
             raise ValueError(
-                "'func' output should have a shape ``(-1, s)`` with ``s`` "
+                "'func' output should have a shape ``(s, -1)`` with ``s`` "
                 "the number of output."
             )
 
-        f_B = np.asarray(func(B))
-        f_AB = np.asarray(func(AB))
+        f_B = np.atleast_2d(func(B))
+        f_AB = np.atleast_2d(func(AB))
```

Comment:
Did you know that the `atleast_` functions accept multiple arrays and return multiple arrays?

Reply:
I did not! I will have a look.